src/util/text-utils.cc

```diff
@@ -188,67 +190,76 @@ inline bool is_nan_text(const std::string &in, const std::string &prefix) {
   return true;
 }
 
-template<typename T>
-bool convert_special_number(const std::string &str, T *out) {
-  if (stricmp(str, "infinity") || stricmp(str, "inf") ||
-      starts_with(str, "1.#INF")) {
```

Comment:
I'm looking at this deleted code and it seems to be using alternative printed representations of nan.
Maybe this is how Window prints them.  I think it would be best to modify the code you've written to call
this function in cases where it's not a regular number, and to extend the test code to support the
cases that this function seems to be trying to handle.
This will make it less likely that we'll experience regressions on Windows.